### DISCUSSION TITLE
Why custom headerLeft does not work the same way when using @react-navigation/native-stack

### DISCUSSION THREAD

**Discussion by roni-castro-db** [react-navigation/react-navigation#10453] [Q&A] [answered]
When using a custom `headerLeft` of `@react-navigation/native-stack` with the flag `initial: false` passed in the navigation it seems to ignore the [initial: false](https://reactnavigation.org/docs/nesting-navigators/#rendering-initial-route-defined-in-the-navigator) flag and resets the initial screen of that stack.

Example of navigation, which custom headerLeft does not work as expected in `@react-navigation/native-stack`:
```js
navigation.navigate('SettingsTab', {
      initial: false,
      screen: 'SettingsDetail',
})
```

https://snack.expo.dev/@ronicesarrc/98ef92

Simulation of the problem:

* Stack `SettingsTab` initialRoute is changed from `SettingsHome` to `SettingsDetail`, when using  `@react-navigation/native-stack` and pressing the headerLeft back button:

https://user-images.githubusercontent.com/95294257/159730616-18652b09-aa45-4db0-bd75-20b9836bdac8.mov

* `SettingsTab` stack initialRoute continues being `SettingsHome` as expected, when using  `@react-navigation/stack`  and pressing the headerLeft back button:

https://user-images.githubusercontent.com/95294257/159730702-cc86a503-1068-4a87-b332-cfb9b5e11353.mov

**Comment by satya164** [comment, marked as answer]
Try updating to latest version of native-stack

**Reply by roni-castro-db** [reply]
It seems to have fixed the problem 🙌 .  Thanks @satya164 
Its behavior is a little bit different from the JS stack one, which opens the previous screen for a short period of time before navigating to the final one, but maybe that's the expected native way.

js stack navigation shows the previous screen for a shorter time: 

https://user-images.githubusercontent.com/95294257/159744790-d0599a9e-dc99-4998-9e3a-a06bd3e6403a.mov

native stack navigation shows the previous screen for a bit more time: 

https://user-images.githubusercontent.com/95294257/159744824-05a0f745-c48f-4d4a-8f94-9add90feb01e.mov
